server/src/main/java/io/crate/types/RegprocType.java

```diff
@@ -56,8 +56,10 @@ public Streamer<Regproc> streamer() {
     public Regproc implicitCast(Object value) throws IllegalArgumentException, ClassCastException {
         if (value == null) {
             return null;
-        } else if (value instanceof Integer) {
-            return Regproc.of((int) value, value.toString());
+        } else if (value instanceof Integer num) {
+            return Regproc.of(num, value.toString());
+        } else if (value instanceof Long num) {
+            return Regproc.of(num.intValue(), value.toString());
```

Comment:
Don't we have to validate the long range to return a meaningful error message instead of silently go to -1? This would align with other casts..